package org.tests;

import org.*;
import org.parser.Parser;

import AST.*;
import AST.Visitor.*;
import Throwables.*;
import java_cup.runtime.Symbol;

import java.io.File;
import java.io.FileNotFoundException;
import java.io.FileReader;
import java.io.Reader;
import java.util.*;

public class ParserTest {
	
	private Reader reader;

	public ParserTest(String reader) {
		this.reader = new java.io.StringReader(reader);
	}

	public ParserTest(File file) {
		try {
			this.reader = new FileReader(file);
		} catch (FileNotFoundException e) {
			e.printStackTrace();
		}
	}
	
	public void run(){
		try {
			            
            Scanner s = new Scanner(this.reader);
            Parser p = new Parser();
            Symbol root;
            root = p.parse();

            Program prog = (Program)root.value;
            prog.accept(new PrettyPrintVisitor());

            System.out.print("\nParsing successfull"); 
        } catch (CompilerException e) {
            // an error in the user's arguments or input, or some
            // other kind of error that we've already handled in the
            // appropriate way (not a bug that the error got here)
            System.err.println(e.getMessage());
        } catch (Exception e) {
            // yuck: some kind of error in the compiler implementation
            // that we're not expecting (a bug!)
            System.err.println("Unexpected internal compiler error: " + 
                               e.toString());
            // print out a stack dump
            e.printStackTrace();
        }	
	}
	
}
